source/modulo_components/modulo_components/component.py

```diff
@@ -94,3 +97,10 @@ def add_output(self, signal_name: str, data: str, message_type: MsgT,
             self._outputs[parsed_signal_name]["publisher"] = publisher
         except Exception as e:
             self.get_logger().error(f"Failed to add output '{signal_name}': {e}")
+
+    def raise_error(self):
+        """
+        Set the in_error_state predicate to true.
+        """
+        super().raise_error()
+        self.set_predicate("in_error_state", True)
```

Comment:
does this also immediately publish the predicate? otherwise it won't go through since step is skipped after. I would also consider if you can cancel the step timer entirely, to save a tiny bit of compute for the executor

Reply:
It does publish immediately yes. If I want to stop the timer I need to expose a helper function from the component interface